ui/static/ui/js/listing_resources.jsx

```diff
@@ -1,7 +1,138 @@
+
+    var getImageFile = function(resourceType) {
+      if (resourceType === 'chapter') {
+        return 'ic-book.png';
+      } else if (resourceType === 'sequential') {
+        return 'ic-sequential.png';
+      } else if (resourceType === 'vertical') {
+        return 'ic-vertical.png';
+      } else if (resourceType === 'problem') {
+        return 'ic-pieces.png';
+      } else if (resourceType === 'video') {
+        return 'ic-video.png';
+      } else {
+        return 'ic-code.png';
+      }
+    };
+
+    var Facet = React.createClass({
+      render: function() {
+        var icon = null;
+        if (this.props.facetId === 'resource_type') {
+          var src = this.props.imageDir + "/" + getImageFile(this.props.id);
+          icon = <span className="min-width">
+            <img src={src} className="ic-custom"/>
+            </span>;
+        }
+
+        return <li>
+          <ICheckbox
+            id={"check-" + this.props.id}
+            tabIndex="add"
+            className="icheck-11" onChange={this.handleChange}
+            checked={this.props.selected} />
+          {icon}
+          <label htmlFor={"check-" + this.props.id}>
+            {this.props.label}
+          </label>
+          <span className="badge">
+            {this.props.count}
+          </span>
+        </li>;
+      },
+      handleChange: function(e) {
+        this.props.updateFacets(
+          this.props.facetId, this.props.id, e.target.checked);
+      }
+    });
+
+    var FacetGroup = React.createClass({
+      render: function() {
+        var thiz = this;
+        var facets = _.map(this.props.values, function(value) {
+          var selected = thiz.props.selectedFacets[value.key];
+
+          return <Facet label={value.label}
+                        id={value.key}
+                        count={value.count}
+                        key={value.key}
+                        facetId={thiz.props.id}
+                        imageDir={thiz.props.imageDir}
+                        updateFacets={thiz.props.updateFacets}
+                        selected={selected}
+            />;
+        });
+
+        var collapseId = "collapse-" + this.props.id;
+
+        return <div className="panel panel-default">
+          <div className="panel-heading">
+            <h4 className="panel-title">
+              <a className="accordion-toggle"
+                 href={"#" + collapseId}
+                 data-parent="#accordion"
+                 data-toggle="collapse"
+                >
+                {this.props.label}
+              </a>
+            </h4>
+          </div>
+          <div className="panel-collapse collapse in" id={collapseId}>
+            <div className="panel-body">
+              <ul className="icheck-list">
+                {facets}
+              </ul>
+            </div>
+          </div>
+        </div>;
+      }
+    });
+
+    var Facets = React.createClass({
+      render: function() {
+        var thiz = this;
```

Comment:
In the `makeFacetGroup` function it's used to refer to the `this` from the render function